bundles/org.eclipse.osgi/supplement/src/org/eclipse/osgi/util/ResourceFormatter.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2012, 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - extracted from ModuleResolutionReport / ModuleRevision
+ *******************************************************************************/
+package org.eclipse.osgi.util;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import org.eclipse.osgi.internal.framework.FilterImpl;
+import org.osgi.framework.Constants;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.namespace.BundleNamespace;
+import org.osgi.framework.namespace.HostNamespace;
+import org.osgi.framework.namespace.PackageNamespace;
+import org.osgi.resource.Capability;
+import org.osgi.resource.Namespace;
+import org.osgi.resource.Requirement;
+
+/**
+ * Utility class that provides some generic Resource API > String conversion
+ * that are suitable to be presented to the user.
+ * 
+ * @since 3.19
+ */
+public final class ResourceFormatter {
```

Comment:
Name seems odd because no methods take a resource.  Perhaps add the static methods to the "main" class in `org.eclipse.osgi.container` package `ModuleContainer` where we have other static helper methods like `createRequirement(String, Map<String, String>, Map<String, ?>)`

All public methods will need javadoc

Reply:
Done!